packages/ds/src/components/select/select.tsx

```diff
@@ -0,0 +1,197 @@
+
+  &::placeholder {
+    color: ${props => props.theme.colors.text.secondary};
+  }
+
+  &:hover {
+    border-color: ${props =>
+      props.error
+        ? props.theme.colors.ui.error
+        : props.theme.colors.ui.primary};
+  }
+
+  &:focus {
+    ${focusRing}
+
+    &::placeholder {
+      color: transparent;
+    }
+  }
+
+  &:-moz-read-only {
+    background-color: ${props => props.theme.colors.ui.tertiary};
+    border-color: ${props => props.theme.colors.ui.secondary};
+  }
+
+  &:read-only {
+    background-color: ${props => props.theme.colors.ui.tertiary};
+    border-color: ${props => props.theme.colors.ui.secondary};
+
+    &::placeholder {
+      color: ${props => props.theme.colors.text.secondary};
+    }
+  }
+
+  &:disabled {
+    -webkit-text-fill-color: currentColor; /* set text fill to current color for safari */
+    opacity: 1; /* correct opacity on iOS */
+    color: ${props => props.theme.colors.text.disabled};
+    background-color: ${props => props.theme.colors.ui.disabled};
+    border-color: ${props => props.theme.colors.ui.disabled};
+
+    &::placeholder {
+      color: ${props => props.theme.colors.text.disabled};
+      opacity: 1;
+    }
+  }
+
+  ${compose(space, layout, flexbox, border, position, color)}
+`;
+
+ContentArea.defaultProps = {
+  pt: selectTokens.y,
+  pb: selectTokens.y,
+  borderWidth: '1px',
+  borderStyle: 'solid',
+  borderRadius: 0,
+  color: 'text.primary',
+  bg: 'bg[1]',
+  mb: 0,
+};
+
+const LeftIcon = styled(Box)<BoxProps & { disabled?: boolean }>`
+  position: absolute;
+
+  svg {
+    display: block;
+    font-size: ${props => props.theme.fontSizes[4]};
+    color: ${props =>
+      props.disabled
+        ? props.theme.colors.text.disabled
+        : props.theme.colors.text.primary};
+  }
+`;
+
+const RightIcon = styled(Box)<BoxProps & { disabled?: boolean, readOnly?: boolean, error?: boolean }>`
+  position: absolute;
+
+  svg {
+    display: block;
+    font-size: ${props => props.theme.fontSizes[4]};
+    ${props => props.disabled && { color: props.theme.colors.text.disabled }};
+    ${props => props.readOnly && { color: props.theme.colors.text.secondary }};
+    ${props => props.error && { color: props.theme.colors.ui.error }};
+  }
+`;
+
+export const Select = forwardRef<HTMLSelectElement, SelectProps>(
+  (
+    { leftIcon, rightIcon, mb, mt, mx, my, ml, mr, disabled, ...props },
+    ref
+  ) => (
+    <Flex
+      alignItems="center"
+      position="relative"
+      mx={mx}
+      my={my}
+      mb={mb}
+      mt={mt}
+      ml={ml}
+      mr={mr}
+    >
+    {leftIcon && (
+      <LeftIcon left={selectTokens.x} disabled={disabled}>
+        {leftIcon}
+      </LeftIcon>
+    )}
+    <ContentArea
+      as="select"
+      variant="body"
+      ref={ref}
+      py={2}
+      hasLeftIcon={!!leftIcon}
+      hasRightIcon={!!rightIcon}
+      disabled={disabled}
+      aria-invalid={props.error ? 'true' : 'false'}
+      {...props}
+    />
+    {rightIcon && (
```

Comment:
The right icon would just always be the `ExpandMore`, right?

Reply:
sorry, never mind. Just saw the figma specs

Reply:
We have two right icons actually. One is `ExpandMore` which doesn't change and second one sits left side of `ExpandMore` and is optional. Designs are here:
https://www.figma.com/file/RqENxZWAzGiEWM7COch1Sc/Radius-Design-Kit?node-id=542%3A6317

I thought `rightIcon` styled-component could be leveraged for both so kept its name. Though, open to suggestions as renaming or restructuring it